### ISSUE TITLE
Swift runtime failure: Double value cannot be converted to UInt16 because the result would be greater than UInt16.max

### ISSUE THREAD

**Issue by osy** [migueldeicaza/SwiftTerm#252]
This line: https://github.com/migueldeicaza/SwiftTerm/blob/main/Sources/SwiftTerm/iOS/iOSExtensions.swift#L15

causes a crash sometimes on iOS 14 with the error: `Swift runtime failure: Double value cannot be converted to UInt16 because the result would be greater than UInt16.max`

It seems to be a precision issue:

```
(lldb) p red
(CoreFoundation.CGFloat) $R3 = (native = 0.99990183115005493)
(lldb) p green
(CoreFoundation.CGFloat) $R4 = (native = 1.0000687837600708)
(lldb) p blue
(CoreFoundation.CGFloat) $R5 = (native = 0.99987989664077759)
```

**Comment by migueldeicaza** [comment]
Whoa, fascinating!  How did you get that green value 1.0000687?

**Comment by migueldeicaza** [comment]
Oh, I see, before iOS 10, the values could only be between 0.0 and 1.0, after iOS 10, if the values are within that range, they are in the sRGB gamut, otherwise, they are outside that range.

Given that these are RGB values for the terminal, I am going to clamp those.

**Comment by migueldeicaza** [comment]
Fixed in f1716ef7f5a6bf40e8c835052831d681bbb232b3

**Comment by osy** [comment]
That makes sense, thanks.
